src/plugins/workspace/public/containers/workspace_dropdown_list/workspace_dropdown_list.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useCallback } from 'react';
+
+import { EuiButton, EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import { useEffect } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { CoreStart, WorkspaceAttribute } from '../../../../../core/public';
+
+type WorkspaceOption = EuiComboBoxOptionOption<WorkspaceAttribute>;
+
+interface WorkspaceDropdownListProps {
+  coreStart: CoreStart;
+  onCreateWorkspace: () => void;
+  onSwitchWorkspace: (workspaceId: string) => void;
+}
+
+function workspaceToOption(workspace: WorkspaceAttribute): WorkspaceOption {
+  return { label: workspace.name, key: workspace.id, value: workspace };
+}
+
+export function WorkspaceDropdownList(props: WorkspaceDropdownListProps) {
+  const { coreStart, onCreateWorkspace, onSwitchWorkspace } = props;
+  const [loading, setLoading] = useState(true);
+  const [currentWorkspace, setCurrentWorkspace] = useState([] as WorkspaceOption[]);
+  const [workspaceOptions, setWorkspaceOptions] = useState([] as WorkspaceOption[]);
```

Comment:
This state is unnecessary too, it can be converted from workspaceList, try to refactor code like these:
```tsx
const [searchValue, setSearchValue] = useState<string>("");
const workspaceOptions = useMemo(()=>{
    return workspaceList.map(workspaceToOption).filter((item) => !searchValue || item.label.includes(searchValue));
}, [workspaceList, searchValue])
```

Reply:
@zhichao-aws It seems that the updated code still need the workspaceOptions state. And we need to init the workspaceOptions when `mount`. The code works fine, but will cause unnecessary re-render. I can approve it first, maybe we can create a new PR to update it in the future.

Reply:
We discuss the code off-line and agree we can keep current design